pkg/asset/installconfig/gcp/client.go

```diff
@@ -343,3 +352,56 @@ func (c *Client) getServiceUsageService(ctx context.Context) (*serviceusage.Serv
 func (c *Client) GetCredentials() *googleoauth.Credentials {
 	return c.ssn.Credentials
 }
+
+func (c *Client) getPermissions(ctx context.Context, project string, permissions []string) ([]string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	service, err := c.getCloudResourceService(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get cloud resource manager service")
+	}
+
+	projectsService := cloudresourcemanager.NewProjectsService(service)
+	rb := &cloudresourcemanager.TestIamPermissionsRequest{Permissions: permissions}
+	response, err := projectsService.TestIamPermissions(project, rb).Context(ctx).Do()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get Iam permissions")
+	}
+
+	return response.Permissions, nil
+
+}
+
+// GetProjectPermissions will consume a set of permissions as strings and return
+// a map of of whether or not the permission was found for the service account in the provided project.
+// A list of valid permissions can be found at https://cloud.google.com/iam/docs/understanding-roles.
+func (c *Client) GetProjectPermissions(ctx context.Context, project string, permissions []string) (map[string]bool, error) {
+	validPermissions, err := c.getPermissions(ctx, project, permissions)
+	if err != nil {
+		return nil, err
+	}
+
+	permissionsMap := make(map[string]bool)
+	for _, perm := range permissions {
+		permissionsMap[perm] = false
+	}
+
+	for _, validPerm := range validPermissions {
+		permissionsMap[validPerm] = true
+	}
+
+	logrus.Warnf("%+v", permissionsMap)
```

Comment:
Maybe
```suggestion
	logrus.Debugf("%+v", permissionsMap)
```
is more appropriate? Or is this temporary just to check it in CI?

Reply:
This was temporary and actually supposed to have been removed already ! I used Warnf as it alerts in Yellow and is easier to find, but this can be removed.